core/lib/via_btc_client/src/inscriber/mod.rs

```diff
@@ -1 +1,734 @@
+            .ok_or_else(|| anyhow::anyhow!("Failed to get witness"))? =
+            Witness::p2wpkh(&fee_payer_signature, &fee_payer_pk);
+
+        // sign tapscript reveal input
+
+        let sighash_type = TapSighashType::All;
+        let prevouts = Prevouts::All(&input.prev_outs);
+
+        let reveal_input_sighash = sighasher
+            .taproot_script_spend_signature_hash(
+                REVEAL_TX_TAPSCRIPT_REVEAL_INDEX as usize,
+                &prevouts,
+                TapLeafHash::from_script(
+                    &inscription_data.inscription_script,
+                    LeafVersion::TapScript,
+                ),
+                sighash_type,
+            )
+            .context("Failed to create sighash")?;
+
+        // Sign the tapscript reveal sighash using the signer
+        let msg = Message::from_digest(reveal_input_sighash.to_byte_array());
+        let reveal_input_signature = self.signer.sign_schnorr(msg)?;
+
+        // verify signature
+        let internal_key = self.signer.get_internal_key()?;
+        let secp_ref = self.signer.get_secp_ref();
+
+        secp_ref
+            .verify_schnorr(&reveal_input_signature, &msg, &internal_key)
+            .context("Failed to verify signature")?;
+
+        // Update the witness stack.
+
+        let reveal_input_signature = bitcoin::taproot::Signature {
+            signature: reveal_input_signature,
+            sighash_type,
+        };
+
+        let mut witness_data: Witness = Witness::new();
+
+        witness_data.push(&reveal_input_signature.to_vec());
+        witness_data.push(&inscription_data.inscription_script.to_bytes());
+
+        // add control block to witness
+        let control_block = input.control_block.clone();
+        witness_data.push(&control_block.serialize());
+
+        *sighasher
+            .witness_mut(REVEAL_TX_TAPSCRIPT_REVEAL_INDEX as usize)
+            .ok_or_else(|| anyhow::anyhow!("Failed to get witness"))? = witness_data;
+
+        let reveal_tx = sighasher.into_transaction();
+
+        let res = FinalTx {
+            tx: reveal_tx.clone(),
+            txid: reveal_tx.compute_txid(),
+        };
+
+        Ok(res)
+    }
+
+    async fn broadcast_inscription(&self, commit: &FinalTx, reveal: &FinalTx) -> Result<bool> {
+        let commit_tx_hex = commit.tx.raw_hex().to_string();
+        let reveal_tx_hex = reveal.tx.raw_hex().to_string();
+
+        let mut commit_broadcasted = false;
+        let mut reveal_broadcasted = false;
+
+        // broadcast commit tx with retry
+        for _ in 0..BROADCAST_RETRY_COUNT {
+            let res = self
+                .client
+                .broadcast_signed_transaction(&commit_tx_hex)
+                .await;
+
+            if res.is_ok() {
+                commit_broadcasted = true;
+                break;
+            }
+        }
+
+        // broadcast reveal tx with retry
+        for _ in 0..BROADCAST_RETRY_COUNT {
```

Comment:
let's change it in more rusty way:
```
let reveal_broadcasted = (0..BROADCAST_RETRY_COUNT).any(|_| {
    match self.client.broadcast_signed_transaction(&reveal_tx_hex).await {
        Ok(_) => true,
        Err(_) => false,
    }
});
```

Reply:
when i did this i should deal to managing async inside of closure (because i need use .await for broadcasting)  and it's make the code complex. Imo the current way is simpler and it's ok.